test MOdule/testOCSP.pl

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env perl
+
+use warnings;
+use strict;
+
+use Socket;
+use Net::SSLeay qw(die_now die_if_ssl_error);
+
+use IO::Socket::SSL qw(debug0);
+
+use Time::gmtime;
+use Time::ParseDate;
+use HTTP::Tiny;
+use Data::Dumper;
+Net::SSLeay::load_error_strings();
+Net::SSLeay::SSLeay_add_ssl_algorithms();
+Net::SSLeay::randomize();
+
+my ($dest_serv, $port) = ("www.migros.ch","443");    # Read command line
+$port = getservbyname( $port, 'tcp' ) unless $port =~ /^\d+$/;
+	my $dest_ip = gethostbyname($dest_serv);
+	my $dest_serv_params = sockaddr_in( $port, $dest_ip );
+
+	socket( S, &AF_INET, &SOCK_STREAM, 0 ) or die "socket: $!";
+	connect( S, $dest_serv_params ) or die "connect: $!";
+	select(S);
+	$| = 1;
+	select(STDOUT);                   # Eliminate STDIO buffering
+
+	# The network connection is now open, lets fire up SSL
+
+	my $ctx = Net::SSLeay::CTX_new() or die_now("Failed to create SSL_CTX $!");
+	Net::SSLeay::CTX_set_options( $ctx, &Net::SSLeay::OP_ALL )
+	  and die_if_ssl_error("ssl ctx set options");
+	my $ssl = Net::SSLeay::new($ctx) or die_now("Failed to create SSL $!");
+	Net::SSLeay::set_fd( $ssl, fileno(S) );    # Must use fileno
+	my $res = Net::SSLeay::connect($ssl) and die_if_ssl_error("ssl connect");
+
+# get id(s) for given certs, like from get_peer_certificate
+# or get_peer_cert_chain. This will croak if
+# - one tries to make an OCSP_CERTID for a self-signed certificate
+# - the issuer of the certificate cannot be found in the SSL objects
+#   store, nor in the current certificate chain
+my $cert = Net::SSLeay::get_peer_certificate($ssl);
+
+my $id = eval { Net::SSLeay::OCSP_cert2ids($ssl,$cert) };
+die "failed to make OCSP_CERTID: $@" if $@;
+
+# create OCSP_REQUEST from id(s)
+# Multiple can be put into the same request, if the same OCSP responder
+# is responsible for them.
+my $req = Net::SSLeay::OCSP_ids2req($id);
+ 
+# determine URI of OCSP responder
+my $uri = Net::SSLeay::P_X509_get_ocsp_uri($cert);
+# Send stringified OCSP_REQUEST with POST to $uri.
+# We can ignore certificate verification for https, because the OCSP
+# response itself is signed.
+my $ua = HTTP::Tiny->new(verify_SSL => 0);
+$res = $ua->request( 'POST',$uri, {
+    headers => { 'Content-type' => 'application/ocsp-request' },
```

Comment:
Hello,

Je te contact car j’ai quelques petits problème avec la compréhension de Catalyst.

J’ai mis en place l’outil sur ma VM et commencé avec les view bootstrap.

J’ai de la peine a comprendre comment et si on peut utiliser les modules Perl que j’ai utilisé pour mon script ou pas.

Si je peux garder ma classe .pm que j’ai créée en Perl et comment interagir avec un fichier XML.

C’est un peut compliqué pour moi qui n’ai jamais fait de web…

Si tu a quelques infos à me donner ça serait vraiment sympa.

Salutations

Behar

> Le 20 août 2015 à 07:01, Jeanneret Cedric <notifications@github.com> a écrit :
> 
> In test MOdule/testOCSP.pl <https://github.com/EthACKdotOrg/swiss-ssl-lawn/pull/2#discussion_r37495366>:
> 
> > +my $id = eval { Net::SSLeay::OCSP_cert2ids($ssl,$cert) };
> > +die "failed to make OCSP_CERTID: $@" if $@;
> > +
> > +# create OCSP_REQUEST from id(s)
> > +# Multiple can be put into the same request, if the same OCSP responder
> > +# is responsible for them.
> > +my $req = Net::SSLeay::OCSP_ids2req($id);
> > + 
> > +# determine URI of OCSP responder
> > +my $uri = Net::SSLeay::P_X509_get_ocsp_uri($cert);
> > +# Send stringified OCSP_REQUEST with POST to $uri.
> > +# We can ignore certificate verification for https, because the OCSP
> > +# response itself is signed.
> > +my $ua = HTTP::Tiny->new(verify_SSL => 0);
> > +$res = $ua->request( 'POST',$uri, {
> > +    headers => { 'Content-type' => 'application/ocsp-request' },
> social links might track the user, hence it's a privacy issue, especially when users are authenticated to the social network (facebook will know you'd at least checked UBS or another bank).
> Might be interesting to check cookies as well: for example ethack does have links to social networks, but they are static, no javascript, hence no external cookie, hence no tracking.
> 
> Regarding redirections, checking where you end is the right way. WhatWeb had some issues last time we used it, but you may give it a try.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/EthACKdotOrg/swiss-ssl-lawn/pull/2/files#r37495366>.
>